source/beamline/BioXAS/BioXASSSRLMonochromatorMaskControl.cpp

```diff
@@ -0,0 +1,197 @@
+#include "BioXASSSRLMonochromatorMaskControl.h"
+
+BioXASSSRLMonochromatorMaskControl::BioXASSSRLMonochromatorMaskControl(const QString &name, QObject *parent) :
+	AMPseudoMotorControl(name, "", parent)
+{
+	// Initialize inherited variables.
+
+	value_ = None;
+	setpoint_ = None;
+	minimumValue_ = Open;
+	maximumValue_ = None;
+
+	setEnumStates(QStringList() << valueToString(Open) << valueToString(Closed) << valueToString(None));
+	setAllowsMovesWhileMoving(false);
+	setContextKnownDescription("BeamStatus");
+
+	// Initialize local variables.
+
+	upperBlade_ = 0;
+	lowerBlade_ = 0;
+	status_ = 0;
+}
+
+BioXASSSRLMonochromatorMaskControl::~BioXASSSRLMonochromatorMaskControl()
+{
+
+}
+
+bool BioXASSSRLMonochromatorMaskControl::canMeasure() const
+{
+	bool result = false;
+
+	if (isConnected())
+		result = (upperBlade_->canMeasure() && lowerBlade_->canMeasure() && status_->canMeasure());
+
+	return result;
+}
+
+//bool BioXASSSRLMonochromatorMaskControl::canMove() const
```

Comment:
Commented out code.  Did you want to remove this @helfrij or is this something that is going to be updated and re-added?